### PR TITLE
Queue uses a mock-able replacement for time package

### PR DESCRIPTION
**Pull request by cdzombak**
## Description

This PR replaces usage of `time.Now` and related functions in `queue` and `queue_processor` by https://github.com/jonboulle/clockwork . It also replaces the (hacky and only partly-correct) `getNow` stub.

## Motivation

This ought to allow more stable and reliable tests of the queue.

## Type of change (choose one)
- [x] Chore (refactors, upgrades, etc.)
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Security fix (non-breaking change that fixes a potential vulnerability)
- [ ] Docs
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## Checklist
- [ ] I've linked any associated issues to this PR.
- [x] I've tested my own changes.

*[Check our Pull Request Guidelines](https://github.com/inngest/inngest/blob/main/docs/PULL_REQUEST_GUIDELINES.md)*

### CODE REVIEW

**Comment by cdzombak** [comment]
Interestingly, only a few tests in `queue_test` actually modified the `getNow` stub. Those have been fixed up to use the new mock; other references were simply replaced by calls to `time.Now` since that's all they were anyway.

File: pkg/execution/state/redis_state/queue.go [hunk 1]
```diff
@@ -6,37 +6,38 @@ import (
...
-	"github.com/redis/rueidis"
-	"github.com/rs/zerolog"
-	"gonum.org/v1/gonum/stat/sampleuv"
-	"lukechampine.com/frand"
 )
 
-var (
-	PartitionSelectionMax int64 = 100
-	PartitionPeekMax      int64 = PartitionSelectionMax * 3
+const (
```
Comment (cdzombak) [inline pkg/execution/state/redis_state/queue.go:38]: Unrelated change; these never seem to be modified (and it's unclear that they could be modified safely at runtime).

**Review by BrunoScheufler — APPROVED** [review]

**Comment by cdzombak** [comment]
I would like very careful review from both @tonyhb and @BrunoScheufler on this PR; in particular, it's important that all usages of `time.Now()` and things that call it (like `time.Since`) get replaced by calls to `q.clock` instead.

(There is one usage of time.Now that I decided is okay for now, in the `duration` helper function. We can always figure out how to inject a clock in there later.)

**Comment by cdzombak** [comment]
I also need to double check that no Lua scripts read the current time independent from Go code.

**Review by tonyhb — APPROVED** [review]
